src/TetherComponent.jsx

```diff
@@ -114,31 +104,83 @@ class TetherComponent extends Component {
   }
 
   getTetherInstance() {
-    return this._tether;
+    return this._tetherInstance;
   }
 
   disable() {
-    this._tether.disable();
+    this._tetherInstance.disable();
   }
 
   enable() {
-    this._tether.enable();
+    this._tetherInstance.enable();
   }
 
   on(event, handler, ctx) {
-    this._tether.on(event, handler, ctx);
+    this._tetherInstance.on(event, handler, ctx);
   }
 
   once(event, handler, ctx) {
-    this._tether.once(event, handler, ctx);
+    this._tetherInstance.once(event, handler, ctx);
   }
 
   off(event, handler) {
-    this._tether.off(event, handler);
+    this._tetherInstance.off(event, handler);
   }
 
   position() {
-    this._tether.position();
+    this._tetherInstance.position();
+  }
+
+  _getTargetRef = targetNode => {
+    this._targetNode = targetNode;
+  };
+
+  _getElementRef = elementNode => {
+    this._elementNode = elementNode;
+  };
+
+  _runRenders() {
+    // To obtain the components, we run the render functions and pass in innerRef
+    // Later, when the component is mounted, the ref functions will be called
+    // and trigger a tether update
+    let targetComponent =
+      typeof this.props.renderTarget === 'function'
+        ? this.props.renderTarget({ innerRef: this._getTargetRef })
```

Comment:
I think as a user of `react-tether` I would question what makes this an "inner" ref. I think from that side it might make more sense to either receive an object with a `ref` property. Or since there's only one thing being passed, you could just pass the it as a regular parameter. What do you think?

Reply:
"innerRef" was used as a name because it's terminology that is used elsewhere, such as in [react-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/guides/using-inner-ref.md) or (now deprecated) in [styled-components](https://www.styled-components.com/docs/api#deprecated-innerref-prop).

The rationale for passing an object is that if anything else needed to be passed through the render props it will be easier to add something thus an easier upgrade path. However this is not likely to happen, so passing ref as a direct argument is probably fine.

@danreeves do you have any opinions on what you'd rather the API surface looks like?